worker.go

```diff
@@ -31,8 +32,14 @@ import (
 type (
 	// Worker represents objects that can be started and stopped.
 	Worker interface {
-		Stop()
+		// Start starts the worker in a non-blocking fashion
 		Start() error
+		// Run is a blocking start and cleans up resources when killed
+		Run()
```

Comment:
I prefer having separate methods in general, boolean arguments determining behavior is not ideal. See https://martinfowler.com/bliki/FlagArgument.html, https://softwareengineering.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior